### PR TITLE
New Report History page

### PR DESCRIPTION
**Pull request by pdcp1**
This closes https://github.com/responsible-ai-collaborative/aiid/issues/2218

### CODE REVIEW

File: site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js:4]: nit: should say `Reports`

**Review by cesarvarela — COMMENTED** [review]
Looks good. I'll wait for the small issues in #2216 to get addressed and merged to check on this one again.

File: site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js
Comment (pdcp1) [reply site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js:4]: Fixed

**Comment by cesarvarela** [comment]
@pdcp1 where is the reports history visible?

**Comment by pdcp1** [comment]
> where is the reports history visible?

@cesarvarela Good question lol. I forgot to add the location here.
The buttons are part of the Report action toolbar:

1. On the report card
image

1. On the discovery report card
image

**Review by cesarvarela — COMMENTED** [review]
Not sure if related but I'm getting an error when trying to save changes to a report:

https://deploy-preview-2221--staging-aiid.netlify.app/cite/edit/?report_number=3193&incident_id=559

image

File: site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js
Comment (pdcp1) [reply site/gatsby-site/cypress/e2e/integration/reportHistory.cy.js:5]: We need the `incident_id` to format the `Back to Report` button URL. This works in the same way as the Report Edit page. If we don't pass the `incident_id` param in the querystring, we have to d a query on the `incidents` collection to know that value.

**Comment by pdcp1** [comment]
@cesarvarela 
> Not sure if related but I'm getting an error when trying to save changes to a report:
> 
> https://deploy-preview-2221--staging-aiid.netlify.app/cite/edit/?report_number=3193&incident_id=559

This is happening on Staging as well 😕
https://staging-aiid.netlify.app/cite/edit/?report_number=3193&incident_id=559

I created an [issue](https://github.com/responsible-ai-collaborative/aiid/issues/2237) to fix as high priority

File: site/gatsby-site/src/pages/cite/history.js
Comment (pdcp1) [reply site/gatsby-site/src/pages/cite/history.js]: Yes, use the `fetchPolicy: 'network-only'` is better.
I removed the `useEffect` function and add the `fetchPolicy: 'network-only'` to the `useQuery`

**Review by cesarvarela — CHANGES_REQUESTED** [review]
I made some edits and then when trying to fetch the Report's history page, It crashed:

image

https://deploy-preview-2221--staging-aiid.netlify.app/cite/history/?report_number=3197&incident_id=561

**Comment by pdcp1** [comment]
> I made some edits and then when trying to fetch the Report's history page, It crashed:

This will be fixed after merging this [PR](https://github.com/responsible-ai-collaborative/aiid/pull/2245) that runs a migration.

**Review by cesarvarela — APPROVED** [review]

**Pull request merged**
